diesel/src/query_builder/mod.rs

```diff
@@ -332,15 +332,6 @@ pub trait AsQuery {
     fn as_query(self) -> Self::Query;
 }
 
-impl<T: Query> AsQuery for T {
-    type SqlType = <Self as Query>::SqlType;
-    type Query = Self;
-
-    fn as_query(self) -> Self::Query {
-        self
-    }
-}
-
```

Comment:
Thanks for working on this ❤️ 

This change would unfortunately break most of diesel's query builder. As a blind guess I would expect that this is also the main fix for the ICE, as this impl implies quite a lot of other concrete impls. (Just have a look what else implements `AsQuery` or `Query` to get a feeling for that). If I can help with concrete information here just reach out.

Reply:
Thanks for taking a look! I was just being a bit lazy. I pushed a commit where I did the same change as I did for `AsExpression`, namely replacing `Self` with `T`, and now the diff should look less problematic.

Reply:
@weiznich If you are familiar with the diesel code base, the best way you can help is probably to remove more and more code from diesel/diesel until the ICE stops from happening. Do it "one line at a time". Then we will know what line that is crucial to making the ICE happen. I have tried this myself but I found the diesel code base to be riddled with dependencies spread across the code base via macros and such, and I wasn't tenacious enough to follow through. But finding a minimal way to reproduce the problem we see would certainly help. That is what https://github.com/rust-lang/rust/issues/93588#issuecomment-1139124795 was talking about too.

Once have such an [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example) it will probably not be hard to fix the root problem.